File: bootstrap/bootstrapper.js
const path = require("path");
const fs = require("fs");
const QBT = require("qbittorrent-api");
const parseMagnet = require("parse-magnet-uri")
const EventEmitter = require("events").EventEmitter

class Bootstrapper extends EventEmitter{
    constructor(){
        super();
        this.qbt = QBT.connect("http://localhost:8080", "admin", "adminadmin")
    }

    checkQbtConnection(){
        return new Promise((resolve, reject)=>{
            this.qbt.version((err, data)=>{
                if (err){
                    console.log(`QBT connection error: ${err}`);
                    reject(err)
                }else{
                    console.log("QBT connection successful");
                    resolve()
                }
            })
        })
    }


    bootstrap(manifestMagnet){

    }

}

module.exports = Bootstrapper

File: bootstrap/bootstrapper.py
import qbittorrentapi

class Status(Enum):
    IDLE = 0
    INIT = 1
    ERROR = 99


class Bootstrapper:
    def __init__(self):
        self.status = Status.IDLE
        self.error = None
        self.qbt = qbittorrentapi.Client(host='localhost:8080', username='admin', password='adminadmin')
        try:

            self.qbt.auth_log_in()
        except qbittorrentapi.LoginFailed as e:
            self.set_error(str(e))


    def get_status(self):
        return self.status

    def bootstrap(self, magnet):

        self.set_error(Status.INIT)


    def set_status(self, status):
        self.status = status

    def set_error(self, msg):
        self.error = msg
        self.status = Status.ERROR
